x-pack/protocol/src/main/java/org/elasticsearch/protocol/xpack/ml/job/process/ModelSnapshot.java

```diff
@@ -0,0 +1,333 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.elasticsearch.protocol.xpack.ml.job.process;
+
+import org.elasticsearch.Version;
+import org.elasticsearch.common.ParseField;
+import org.elasticsearch.common.xcontent.ObjectParser;
+import org.elasticsearch.common.xcontent.ObjectParser.ValueType;
+import org.elasticsearch.common.xcontent.ToXContentObject;
+import org.elasticsearch.common.xcontent.XContentBuilder;
+import org.elasticsearch.protocol.xpack.ml.job.Job;
+
+import java.io.IOException;
+import java.util.Date;
+import java.util.Objects;
+
+/**
+ * ModelSnapshot Result POJO
+ */
+public class ModelSnapshot implements ToXContentObject {
+    /**
+     * Field Names
+     */
+    public static final ParseField TIMESTAMP = new ParseField("timestamp");
+    public static final ParseField DESCRIPTION = new ParseField("description");
+    public static final ParseField SNAPSHOT_DOC_COUNT = new ParseField("snapshot_doc_count");
+    public static final ParseField LATEST_RECORD_TIME = new ParseField("latest_record_time_stamp");
+    public static final ParseField LATEST_RESULT_TIME = new ParseField("latest_result_time_stamp");
+    public static final ParseField QUANTILES = new ParseField("quantiles");
+    public static final ParseField RETAIN = new ParseField("retain");
+    public static final ParseField MIN_VERSION = new ParseField("min_version");
```

Comment:
I am on the fence about this value. It could be useful for users who want to move snapshots around, though, it feels like it could just be an internal thing.

Reply:
`min_version` is the earliest version of the product that can read the model snapshot. This is for the autodetect process to protect an older version trying to read model state with new features it is isn't aware of. For informational purposes only and shouldn't be set by the client.

We don't have any APIs that accept a `ModelSnapshot` doc - update and revert use the ID- so I think we should leave this in.

Reply:
I've noticed this field isn't in 6.x so please remove from the backport